### ISSUE TITLE
Auth.currentUserPoolUser() swallows NotAuthorizedException

### ISSUE THREAD

**Issue by thundermiracle** [aws-amplify/amplify-js#3594]
**Describe the bug**
Auth.currentAuthenticatedUser() calls currentUserPoolUser() which swallows 'Access Token has been revoked' error in [THIS LINE](https://github.com/aws-amplify/amplify-js/blob/d384e5dd01b2719111c5121538259571730b1218/packages/auth/src/Auth.ts#L985)  instead of throwing error but resolve(user) which makes bypassCache parameter useless.

**To Reproduce**
1. Use Cognito as SSO service and sign in(SiteA, SiteB)
2. Sign out from SiteA
3. Go to SiteB and use Auth.currentUserPoolUser() or Auth.currentAuthenticatedUser() to check login status
4. console.log throws `POST https://cognito-idp.us-east-2.amazonaws.com/ 400` error, but both currentUserPoolUser(), currentAuthenticatedUser() return user as if it's still logged in

**Expected behavior**
currentUserPoolUser() should throw error and clear localStorage if user is already signed out despite the existence of localStorage

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Chrome
 - Version  75

**Comment by alansfyeung** [comment]
In our backend we use _AdminUserGlobalSignOut_ to revoke access/refresh tokens for a Cognito user (noting that ID tokens can't be revoked this way). Using the _bypassCache_ option with `Auth.currentAuthenticatedUser()` should work in theory to detect revoked sessions, but doesn't as described by the OP. 

In the meantime, _getPreferredMFA()_ is a workaround that we have used to detect revoked sessions. 

```
Auth.currentAuthenticatedUser().then(user => {
  Auth.getPreferredMFA(user, { bypassCache: true }).then(() => user);
}).catch(err => {
  console.error('Session potentially revoked', err);
});
```

**Comment by thundermiracle** [comment]
@alansfyeung Thanks for your comment.
After reading the source code, we're using `currentUserInfo({bypassCache: true})` to detect the sessions.

**Comment by ericclemmons** [comment]
@thundermiracle I was researching this issue today, but am having trouble reproducing.

With two windows open & logged in, after signing out of Window A, whether refreshing the page or explicitly calling `Auth.currentAuthenticatedUser()`, Window B returns the error `not authenticated`.

(This happened whether I passed `{ bypassCache: true }` or not).

Is this still happening for you?

**Comment by thundermiracle** [comment]
@ericclemmons 
Thanks for your reply.
I'm not sure of your reproducing environment, but I guess that you started only 1 APP and opened it in two tab. In that case the library works fine.

The problem is happening in SSO which needs two APPs.
1. Login in App1.
2. App2 can login by calling federatedSignIn()
3. Logout from App1.
4. in App2, call currentAuthenticatedUser() to check the session. -> *failed here*

By the way, I don't think the library can check SSO's session without passing ```{ bypassCache: true }```.

**Comment by ericclemmons** [comment]
@thundermiracle You're exactly right: I misunderstood SiteA/SiteB for being new windows, not different apps entirely.

I'll give this another go...

**Comment by thundermiracle** [comment]
@ericclemmons 
I opened a PR to fix this one.
I'll add some tests if it's acceptable.

But I don't know why currentUserPoolUser() have to swallow all errors except the following two?
```js
// Make sure the user is still valid
if (
  err.message === 'User is disabled' ||
  err.message === 'User does not exist.'
) {
  rej(err);
```

**Comment by ericclemmons** [comment]
Closed via #4281.  Please let us know if this issue isn't resolved.
